sematic/graph.py

```diff
@@ -532,7 +552,8 @@ def clone_futures(self, reset_from: Optional[RunID] = None) -> ClonedFutureGraph
             and future.kwargs appropriately.
 
         reset_from: Optional[str]
-            Force reset other runs than only failed ones.
+            Force reset descendant/downstream runs if this is not None. If it is None,
+            will use the CONTINUE `RerunMode`.
```

Comment:
Why is `RerunMode` needed if we have `reset_from is None` to indicate to continue from failed?

Reply:
We need this because `rerun_id=None` means "don't do rerun from here" in the resolvers:

```
resolver = CloudResolver()  # instantiate a resolver that is not rerunning
resolver = CloudResolver(rerun_from=run_id)  # instantiate a resolver that is rerunning from run_id
resolver = CloudResolver(
    rerun_from=root_run_id,
    rerun_mode=RerunMode.CONTINUING,
)  # instantiate a resolver that is continuing from where a previous one left off
```

Reply:
Ok so IIUC this is to differentiate between `rerun_from is None` because no values were passed, and `rerun_from is None` because `None` was passed.

I believe this can be solved more elegantly without adding a new parameter. Three options:

1. add a `kwargs` to the resolver's init and then test `if "rerun_from" in kwargs`
2. use a different default then `None` for `rerun_from`, e.g. an `_UNSET` constant. This pattern is actually used in several locations in the standard library.
3. make `rerun_from` either a run ID or `Literal["CONTINUE"]`

```python
CONTINUE = "CONTINUE"

class Resolver:
    def __init__(self, ..., rerun_from: Optional[str] = None):

...

if rerun_from is CONTINUE:  # note the `is` instead of `==` for added safety.
```

All three options are lighter than `RerunMode` as they are fairly local changes and do not affect deeper code too much. I personally prefer option 3 as it accommodates the behavior in the `/schedule` endpoint with a single argument as well.